src/pageManager.js

```diff
@@ -705,12 +705,29 @@ export async function initialize() {
     // Register the pageManager content script for all URLs permitted by the extension manifest.
     const matchPatterns = permissions.getManifestOriginMatchPatterns();
 
-    // TODO register this content script so it may be unloaded later. 
+    // Firefox only supports this as of version 105, remove this check when that version of Firefox ships.
+    let persistAcrossSessions = true;
+    const browserInfo = browser.runtime && browser.runtime.getBrowserInfo && await browser.runtime.getBrowserInfo();
+    if (browserInfo && browserInfo.name === "Firefox") {
+        persistAcrossSessions = false;
+    }
+
+    const contentScriptId = "pageManager";
+    let scripts = await browser.scripting.getRegisteredContentScripts({
+        ids: [contentScriptId],
+    });
+
+    if (scripts.length > 0) {
```

Comment:
oh - I was thinking it makes more sense to do
```
if (scripts.length === 0) registerContentScripts()
else doNothing();
```
because if scripts.length > 0 then the script is already registered i.e. we're done.
rather than unregistering and re-registering.

Reply:
just feels like it's cleaner, more performant, and less opportunity for things to go wrong. Especially since the point of this is just to silence an error; would be good to avoid changing functionality as much as possible.

Reply:
Oh I see, so don't bother un/re-registering? That makes sense... let's re-test before we push to make sure it wasn't an accidental fix :)

Reply:
Oh wait I see how I can accommodate that

Reply:
OK I have it store the contentScriptId now instead of a reference to the content script, and it uses that when unregistering. I noticed that `pageManager` doesn't seem to implement this at all, I filed https://github.com/mozilla-rally/web-science/issues/315 to follow up on that.